cmd/serverless/main.go

```diff
@@ -391,6 +395,17 @@ func runAgent(stopCh chan struct{}) (daemon *serverless.Daemon, err error) {
 	return
 }
 
+func buildGlobalTagsMap(functionARN string, functionName string, region string, AWSAccountID string) map[string]string {
```

Comment:
I think we can keep the logic in this PR, but maybe move the buildGlobalTagsMap function to the same file as arn.go, so the refactor is a little more obvious.

Reply:
I would actually be against moving buildGlobalTagsMap to arn.go, this map is something very specific to tracing.

Reply:
My overall thought is that this seems like an area (ARN parsing, setting default tags on the different types of telemetry, etc.) that could use some cleanup. I actually think it would be good to put some time into it now, in the spirit of "leaving the code better than you found it".

Also, ideally we would only parse the ARN once and then store the parsed ARN pieces somewhere so that they could be used for different things -- but I think it's also probably OK to parse the ARN multiple times using a utility function. I would think that the performance cost of that would be negligible, especially if we only need to parse during cold starts and not every single invocation.

Since we are up against a release deadline, I also think it would be sensible to merge this PR close to as-is and then follow up with some refactoring. Just so that this functionality makes it into the release. Even though the current code structure isn't ideal, it's working and tested.